ports/esp32/mphalport.c

```diff
@@ -75,7 +81,15 @@ void check_esp_err(esp_err_t code) {
             return;
         }
         o_str->base.type = &mp_type_str;
+        #if MICROPY_ERROR_REPORTING > MICROPY_ERROR_REPORTING_NORMAL
+        char err_msg[64];
+        esp_err_to_name_r(code, err_msg, sizeof(err_msg));
+        char str[512]; // 256 bytes size leeds to abnormal output
+        snprintf(str, sizeof(str), "0x%04X %s in function '%s' at line %d in file '%s'", code, err_msg, func, line, file);
+        o_str->data = (const byte *)str;
```

Comment:
This will not work because the str object is taking a pointer to the C stack, and the data on the C stack will not be valid once this function returns.  Instead, the string data must be dynamically allocated here.  Probably easiest to use `vstr_printf` for that.